fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java

```diff
@@ -306,6 +307,10 @@ private void handleQuery(MysqlCommand mysqlCommand) {
         if (mysqlCommand == MysqlCommand.COM_QUERY && ctx.getSessionVariable().isEnableNereidsPlanner()) {
             try {
                 stmts = new NereidsParser().parseSQL(originStmt);
+            } catch (NotSupportedException e) {
```

Comment:
why must add new exception type?

Reply:
The `NotSupportedException` is for `ConnectProcessor.handleQuery` to distinguish this case: enable_decimal is not set to `true` and type `Decimal256 ` is met in the query. Or else it will fallback to old planner.